### PR TITLE
Create the Initial Group Rules Engine

### PR DESCRIPTION
**Pull request by blueandgold**
This is the initial work to implement the Group Rules Engine.  What this PR does so far is the first step of creating the rules engine, create the resources, and build the rule book.

Other steps (getting the GCP group data, and finding violations) will follow.
#202

### CODE REVIEW

File: google/cloud/security/common/gcp_type/resource_util.py [hunk 1]
```diff
@@ -42,6 +43,11 @@ class ResourceUtil(object):
             'class': project.Project,
             'plural': 'Projects',
             'level': 1
+        },
+        resource.ResourceType.GROUP: {
+            'class': group.Group,
+            'plural': 'Groups',
+            'level': 4
```
Comment (blueandgold) [inline google/cloud/security/common/gcp_type/resource_util.py]: @carise , what is "level" here?  it looks sequential to me, and i just picked the next number

Comment (carise) [inline google/cloud/security/common/gcp_type/resource_util.py]: I think that was the hierarchy level, but I don't recall if I am even using it. We can revisit it later.

File: google/cloud/security/scanner/samples/group_rules.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+rules:
+  # Only allow Organization users to be in the Organization groups.
+  - name: sample whitelist for group users
+    mode: whitelist
+    resource:
+      - type: group
+        applies_to: self_and_children
+        resource_ids:
+          - my_customer
```
Comment (carise) [inline google/cloud/security/scanner/samples/group_rules.yaml]: Just a question -- do you build the group "hierarchy" and it all rolls up to "my_customer"?

Comment (blueandgold) [inline google/cloud/security/scanner/samples/group_rules.yaml]: Hmmm, I'm not sure how this works yet, downstream when we start to gather the gcp group data.  I assumed this was the initial entry point, to start gathering all the group.  So to that effect, I guess it is the "root" where everything rolls up to?

Comment (carise) [inline google/cloud/security/scanner/samples/group_rules.yaml]: sgtm

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as the PR author is a googler at the time of this PR was created.

**Pull request merged**